modules/vpc-endpoints/main.tf

```
@@ -13,7 +13,7 @@ data "aws_vpc_endpoint_service" "this" {
 
   service         = try(each.value.service, null)
   service_name    = try(each.value.service_name, null)
-  service_regions = try([each.value.service_region], null)
+  service_regions = try(each.value.service_region != null ? [each.value.service_region] : null, null)
```

```suggestion
  service_regions = one([each.value.service_region])
```

Much simpler, but please test it. :)

---

Thanks for the feedback. So I tested 
```hcl
service_regions = one([each.value.service_region])
```
this does not quite work. If  the `endpoints` maps' `service_region` value is a valid string, it evaluates just to the string, but the data sources' `service_regions` expects a list of strings. Further, if `service _region` is not included in the `endpoints` map, it also fails.

---

I think we could change it to this which is a bit easier to understand and read and I successfully tested for all 3 cases (`service_region` not included in the `endpoints` map,  `service_region=null` or  `service_region="valid-region-string"`:
```hcl
service_regions = try(each.value.service_region, null) != null ? [each.value.service_region] : null
```

---

I think this is going to work as expected:

```hcl
service_regions = try(coalescelist(compact([each.value.service_region])), null)
```

Let's use it if it does.

---

It does indeed, thank you! Updated the PR accordingly.

---

Thank you for fixing an issue!